src/engine/behavior_script.c

```diff
@@ -820,13 +820,12 @@ void cur_obj_update(void) {
     BhvCommandProc bhvCmdProc;
     s32 bhvProcResult;
 
-    s32 inRoom = is_mario_in_room();
+    s32 inRoom = cur_obj_is_mario_in_room();
 
-    if (!(objFlags & OBJ_FLAG_PROCESS_OUTSIDE_ROOM)) {
-        if (inRoom == MARIO_OUTSIDE_ROOM) {
-            cur_obj_enable_disable_room_rendering(MARIO_OUTSIDE_ROOM);
-            return;
-        }
+    if (!(objFlags & OBJ_FLAG_PROCESS_OUTSIDE_ROOM)
+     && (inRoom == MARIO_OUTSIDE_ROOM)) {
+        cur_obj_disable_rendering_in_room();
+        return;
```

Comment:
revert formatting please, my PR will conflict with it

Reply:
No longer conflicts, also I flipped them around so that the faster check is done first